### PR TITLE
Documented ingeter migration to spread-minimizing tokens

### PR DESCRIPTION
**Pull request by duricanikolic**
#### What this PR does
This PR adds a documentation on how to migrate ingesters to `spread-minimizing` tokens.

#### Which issue(s) this PR fixes or relates to

Part of #4736 

#### Checklist

- [ ] Tests updated.
- [x] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
LGTM from technical point of view, this matches my understanding and the doc I've reviewed before.

**Comment by knylander-grafana** [comment]
@duricanikolic Thank you for submitting some documentation. i checked out your PR to do a local build. The page isn't showing up in the local build. I'm going to troubleshoot locally and hopefully fix the issue.

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+aliases:
+  - ../operators-guide/configure/configure-spread-minimizing-tokens/
+description: Learn how to migrate ingesters to spread-minimizing tokens.
+weight: 115
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:5]: I added a page weight to control where this page appears in the left table of contents. With this change, Spread-minimizing tokens appears after Zone-aware replication. 
image

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+menuTitle: Spread-minimizing tokens
+title: Migrate ingesters to spread-minimizing tokens
+---
+
+# Migrate ingesters to spread-minimizing tokens
+
+Using this guide, you can configure Mimir's ingesters to use the _spread-minimizing token generation strategy_ .
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:12]: You have a good open. I tried to remove the passive voice. What do you think? Also, maybe include a sentence about why someone would want to use this and when. 

The suggestion below just removes an extra space before the period. 
```suggestion
Using this guide, you can configure Mimir's ingesters to use the _spread-minimizing token generation strategy_.
```

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+## Before you begin
+
+The ingester time series replication should be [configured with enabled zone-awareness](https://grafana.com/docs/mimir/latest/configure/configure-zone-aware-replication/#configuring-ingester-time-series-replication).
+
+{{% admonition type="note" %}}Use spread-mimizing tokens are recommended if [shuffle sharding](https://grafana.com/docs/mimir/latest/configure/configure-shuffle-sharding/#ingesters-shuffle-sharding) is disabled in your ingesters, or, if shuffle-sharding is enabled, but most of the tenants of your system use all available ingesters.
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:18]: I replaced the notes with the admonition shortcode in your document.  

Notes, tips, warnings, and caution admonitions appear with special formatted boxes when rendered. The [admonition shortcode](https://grafana.com/docs/writers-toolkit/write/shortcodes/) tells Hugo to set this paragraph apart with different formatting. 

Here's what the above admonition shortcode looks like in the doc preview: 
image

Comment (knylander-grafana) [reply docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:18]: and I found another mistake in this sentence. Corrected below. 
```suggestion
{{% admonition type="note" %}}Spread-mimizing tokens are recommended if [shuffle sharding](https://grafana.com/docs/mimir/latest/configure/configure-shuffle-sharding/#ingesters-shuffle-sharding) is disabled in your ingesters, or, if shuffle-sharding is enabled, but most of the tenants of your system use all available ingesters.
```

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+{{% /admonition %}}
+
+For simplicity, let’s assume that there are three configured availability zones named `zone-a`, `zone-b`, and `zone-c`. Migration to the _spread-minimizing token generation strategy_ is a complex process performed zone by zone to prevent any data loss.
+
+## Step 1: Disable write requests to ingesters from zone-a
+
+To disable write requests, configure the following flag on the distributor and the ruler:
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:25]: I removed most of the uses of "in order to" and instead used just "to".

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      route="/cortex.Ingester/Push"}[5m]
+  )
+)
+```
+
+You should see something like this:
+
+![No More Write Requests](no-more-write-requests.png)
+
+## Step 2: Shut down ingesters from zone-a
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:49]: I missed this. Zone-a should have formatting, same as you've used in the text. 
```suggestion
## Step 2: Shut down ingesters from `zone-a`
```

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+Next, ensure that all the in-memory series of all ingesters from `zone-a` have been flushed to long-term storage, as well as that the ingesters from `zone-a` have been forgotten from the ring.
+To do this, invoke the [/ingester/shutdown](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/references/http-api/index.md#shutdown) endpoint on all the ingesters from `zone-a`.
+
+Before proceeding to the next step, ensure that all the calls completed successfully completed.
+
+## Step 3: Enable spread-minimizing tokens for ingesters in zone-a
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:56]:
```suggestion
## Step 3: Enable spread-minimizing tokens for ingesters in `zone-a`
```

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+{{% admonition type="note" %}}
+The example uses `zone-a,zone-b,zone-c` to denote a comma-separated list of configured availability zones.
+{{% /admonition %}}
+
+Before proceeding to the next step, ensure that all the ingester pods related to `zone-a` are up and running with the new configuration.
+
+### Optional step: In-order registration of ingesters
+
+Mimir can force the ring to perform an in-order registration of ingesters, for example, to allow an ingester to register its tokens within the ring only after all previous ingesters (with ID lower than its own ID) have already been registered.
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:74]: Lines 74-76 had some run-on sentences with some complicated ideas. I tried to split them into several sentences. Does this look okay?

**Review by knylander-grafana — APPROVED** [review]
I made some wording changes and marked what I did in the PR. I also found some additional mistaskes that I introduced. I've left them as suggestions so you can see where I made changes. 

Please let me know if you have any questions. I'm happy to explain.

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+## Before you begin
+
+The ingester time series replication should be [configured with enabled zone-awareness](https://grafana.com/docs/mimir/latest/configure/configure-zone-aware-replication/#configuring-ingester-time-series-replication).
+
+{{% admonition type="note" %}}Use spread-mimizing tokens are recommended if [shuffle sharding](https://grafana.com/docs/mimir/latest/configure/configure-shuffle-sharding/#ingesters-shuffle-sharding) is disabled in your ingesters, or, if shuffle-sharding is enabled, but most of the tenants of your system use all available ingesters.
+{{% /admonition %}}
+
+For simplicity, let’s assume that there are three configured availability zones named `zone-a`, `zone-b`, and `zone-c`. Migration to the _spread-minimizing token generation strategy_ is a complex process performed zone by zone to prevent any data loss.
+
+## Step 1: Disable write requests to ingesters from zone-a
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:23]:
```suggestion
## Step 1: Disable write requests to ingesters from `zone-a`
```

File: docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+The example uses `zone-a,zone-b,zone-c` to denote a comma-separated list of configured availability zones.
+{{% /admonition %}}
+
+Before proceeding to the next step, ensure that all the ingester pods related to `zone-a` are up and running with the new configuration.
+
+### Optional step: In-order registration of ingesters
+
+Mimir can force the ring to perform an in-order registration of ingesters, for example, to allow an ingester to register its tokens within the ring only after all previous ingesters (with ID lower than its own ID) have already been registered.
+Enabling this feature minimizes a probability that a write request that should be handled by an ingester actually arrives to the ring before the ingester is registered within the ring. In this case, the request gets handled by another ingester.
+This situation that would introduce some deviation from an optimal load distribution.
```
Comment (knylander-grafana) [inline docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md:76]:
```suggestion
This situation could introduce some deviation from an optimal load distribution.
```

**Comment by duricanikolic** [comment]
> I made some wording changes and marked what I did in the PR. I also found some additional mistaskes that I introduced. I've left them as suggestions so you can see where I made changes.
> 
> Please let me know if you have any questions. I'm happy to explain.

Thank you very much @knylander-grafana. I have applied your suggestions, and will merge the PR.

**Pull request merged**